packages/asc-ui/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -5,24 +5,43 @@ import MenuList from './ContextMenuList'
 import { KeyboardKeys } from '../../types'
 import ownerDocument from '../../utils/ownerDocument'
 
+const selectedChildInitial = -1
+
 type Props = {
   position?: AscCore.ContextMenuTypes.Position
   label?: string
   icon?: React.ReactNode
+  open?: boolean
 }
 
-type State = {}
-
-const selectedChildInitial = -1
+type State = {
+  open?: boolean
+  selectedChild: number
+}
 
 class ContextMenu extends React.Component<Props, State> {
-  state = {
+  static defaultProps = {
     open: false,
-    selectedChild: selectedChildInitial,
   }
 
-  list = React.createRef<HTMLDivElement>()
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      open: props.open,
+      selectedChild: selectedChildInitial,
+    }
+  }
 
+  componentDidUpdate(prevProps: Props) {
+    const { open } = this.props;
+
+    if (prevProps.open !== open) {
+      this.setState({ open })
```

Comment:
I get this linting error:
Do not use setState in componentDidUpdateeslint(react/no-did-update-set-state)

Reply:
Replaced with a function